app/src/behaviors/behavior_mod_morph.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_mod_morph
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <zmk/behavior.h>
+
+#include <zmk/matrix.h>
+#include <zmk/endpoints.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/keycode-state-changed.h>
+#include <zmk/events/modifiers-state-changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+struct behavior_mod_morph_config {
+    struct zmk_behavior_binding normal_binding;
+    struct zmk_behavior_binding morph_binding;
+    zmk_mod_flags mods;
+};
+
+struct behavior_mod_morph_data {
+    struct zmk_behavior_binding *pressed_binding;
+};
+
+static int on_mod_morph_binding_pressed(struct zmk_behavior_binding *binding,
+                                        struct zmk_behavior_binding_event event) {
+    const struct device *dev = device_get_binding(binding->behavior_dev);
+    const struct behavior_mod_morph_config *cfg = dev->config;
+    struct behavior_mod_morph_data *data = dev->data;
+    if (zmk_hid_get_mods() & cfg->mods) {
+        data->pressed_binding = (struct zmk_behavior_binding *)&cfg->morph_binding;
```

Comment:
This means, theoretically, that you can't have same instance bound and pressed simultaneously.

I suppose that's ok, but maybe document this?

Reply:
I'm not sure what you mean with "can't have the same instance bound and pressed simultaneously".